### PR TITLE
Refactor RTC code into a single driver for all platforms

### PR DESCRIPTION
**Pull request by crc-32**
`tintin` RTC does not function properly yet, this is simply for refactoring purposes.
https://github.com/pebble-dev/RebbleOS/issues/46

### CODE REVIEW

File: hw/drivers/stm32_rtc/stm32_rtc.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * Author: Barry Carter <barry.carter@gmail.com>
+ * Refactored by Thomas England (crc-32) <thomas.c.england@gmail.com>
+ */
+
+// make sure we use the external osc
+#ifndef __STM32_RTC_H
+#define __STM32_RTC_H
+
+#if defined(STM32F4XX)
```
Comment (jwise) [inline hw/drivers/stm32_rtc/stm32_rtc.h]: I *think* these includes should not be necessary in this file

File: hw/drivers/stm32_rtc/stm32_rtc.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#    include "stm32f2xx.h"
+#else
+#    error "I have no idea what kind of stm32 this is; sorry"
+#endif
+#include <time.h>
+#include "rebble_time.h"
+
+void rtc_init(void);
+void rtc_config(void);
+void hw_get_time_str(char *buf);
```
Comment (jwise) [inline hw/drivers/stm32_rtc/stm32_rtc.h]: get_time_str can die I think

File: hw/platform/tintin/tintin.c [hunk 3]
```diff
@@ -17,6 +17,7 @@
 #include <misc.h>
 
 #include "stm32_power.h"
+#include "tintin_rtc.h"
```
Comment (jwise) [inline hw/platform/tintin/tintin.c]: missing file?

File: hw/drivers/stm32_rtc/stm32_rtc.h (hunk 1 above)
Comment (crc-32) [reply hw/drivers/stm32_rtc/stm32_rtc.h]: Does not build without, I thought the same

File: hw/platform/tintin/tintin.c (hunk 3 above)
Comment (crc-32) [reply hw/platform/tintin/tintin.c]: leftover from test implementation, fixing

**Comment by jwise** [comment]
Merged at master.  Thanks!

**Pull request closed without merging**